system/include/emscripten/val.h

```diff
@@ -402,6 +405,13 @@ class val {
     return handle;
   }
 
+  // Takes ownership of the handle away from, and invalidates, this instance.
+  EM_VAL take_handle() {
```

Comment:
Nit: I'd call this `release_handle()` or `leak_handle()`. `take_handle` usually implies taking ownership of an existing handle (the opposite direction).

Reply:
SGTM, consistent with std::unique_ptr::release()

Reply:
I know I'm nitpicking on my own suggestion now, but I remembered that the function that takes handle is currently called `val::take_ownership`, so maybe this should even be `release_ownership` for symmetry.

OTOH `release_handle` is also fine, I don't have a strong preference.

Reply:
No worries, we'll have to live with the name, good to choose well at the start.